### DISCUSSION TITLE
[Feature] Mark disk for caching

### DISCUSSION THREAD

**Discussion by struthio** [moosefs/moosefs#537] [Ideas]
hi,

Chunkserver uses mfshdd.cfg to configure where it store its data.
How about adding new marking there, marking disk as 'cache' - in understanding that when data in uploaded to chunkserver it is saved on 'cache' disk, and after successful upload it can be redistributed to other disks.

We could then have server with X HDD disks, and Y SSD. SSD could be marked as 'cache', so upload will be fast (as data will be stored in SSD) and then after data is 'safe' it will be distributed to 'permanent/slower' disks.

Currently such solution requires dedicated chunkserver with only SSD drives and using 'CREATE' storage class, and after uploading file, it is using network to redistribute data.

**Comment by oszafraniec** [comment]
No need for this, use chunkserver labels and classes for it like we used to do ;) 

https://moosefs.com/Content/Downloads/moosefs-storage-classes-manual.pdf

<img width="1430" alt="Zrzut ekranu 2023-05-9 o 13 50 35" src="https://github.com/moosefs/moosefs/assets/3938516/05580441-e48c-4b06-ba3d-b707a2dbc5a5">

**Reply by oszafraniec** [reply]
Sorry, didnt read all of your post :/ But you can still start 2 chunkserver processes on one machine, one for HDDs and one for SSDs. Also you can use FS with SSD caching/journaling etc.

**Reply by struthio** [reply]
Yes I could go with storage class as Your 'instant_cache', but issue I had with such is when You upload multiple large files.
Let's assume we upload at once 100 files 10GB each. When we upload first file it goes pretty fast (as it goes directly to SSD and can utilize full network bandwidth of server). In the moment we are done with first file and start uploading second, network bandwidth is cut in half, as in the same time we are uploading second file and MooseFS starts replication of first one between nodes.

Having 'cache drive' directly in chunkserver replication will happen 'locally' in sort of 'internal re-balancing' therefor not impacting network performance.
